src/GraphQL/TypeApi.hs

```diff
@@ -50,17 +50,23 @@ queryError = throwM . QueryError
 -- NormalizedSelectionSet which has query fragments etc. resolved.
 class (MonadThrow m, MonadIO m) => HasGraph m a where
   type HandlerType m a
-  buildResolver :: HandlerType m a -> AST.SelectionSet -> m GValue.Value
+  buildResolver :: HandlerType m a -> CanonicalQuery -> m GValue.Value
 
--- Parse a value of the right type from an argument
--- TODO
+-- | The ReadValue instance converts AST.Value types like ValueInt to
+-- the type expected by the handler function. It's the boundary
+-- between incoming types and your custom application Haskell types.
 class ReadValue a where
```

Comment:
Not for this PR.

I would like us to change this to think a bit about terminology before doing a publicly announced release.

In particular, I think we should use `ToValue` and `FromValue` (or some variation thereof) to parallel aeson.